File: app/src/main/java/com/example/dictionaryapp/database/DictDao.java
package com.example.dictionaryapp.database;

import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;
@Dao
public interface DictDao {

    @Query("SELECT * FROM dictionary WHERE KANJI = :query AND DICTNAME NOT IN (:disabled)")
    public List<DictEntry> searchEntryByKanji(String query, List<String> disabled);

    @Query("SELECT * FROM dictionary WHERE READING = :query AND DICTNAME NOT IN (:disabled)")
    public List<DictEntry> searchEntryReading(String query, List<String> disabled);


}

File: app/src/main/java/com/kamui/rin/database/DictDao.kt
package com.kamui.rin.database

import androidx.room.Dao
import androidx.room.Query

@Dao
interface DictDao {
    @Query("SELECT * FROM dictionary WHERE kanji = :query AND dictname NOT IN (:disabled)")
    fun searchEntryByKanji(
        query: String,
        disabled: List<String?>
    ): List<DictEntry>

    @Query("SELECT * FROM dictionary WHERE reading = :query AND dictname NOT IN (:disabled)")
    fun searchEntryReading(
        query: String,
        disabled: List<String>
    ): List<DictEntry>
}